index.js

```diff
@@ -18,23 +18,78 @@ colors.def = {h: 10, s: 66, l: 80}
 colors.js = {h: 10, s: 66, l: 80}
 colors.c = {h: 10, s: 66, l: 80}
 
+var css = `
+  .d3-flame-graph .frame {
+    transition: transform 500ms ease-in-out;
+  }
+
+  .d3-flame-graph .frame rect {
+    cursor: pointer;
+    transition: width 500ms ease-in-out;
+  }
+
+  .d3-flame-graph .frame foreignObject {
+    overflow: hidden;
+    pointer-events: none;
+    transition: width 500ms ease-in-out;
+  }
+
+  .d3-flame-graph .frame div {
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    font-size: 12px;
+    font-family: Verdana;
+    margin-left: 4px;
+    margin-right: 4px;
+    line-height: 1.5;
+    padding: 0;
+    font-weight: 400;
+    color: #000;
+  }
+`
+
+function eventPath (event) {
+  if (event.path) return event.path
+  if (event.composedPath) return event.composedPath()
+  var target = event.target
+  var path = [target]
+  while ((target = target.parentElement)) {
+    path.push(target)
+  }
+  path.push(document, window)
+  return path
+}
+
 function flameGraph (opts) {
-  var tree = opts.tree 
+  var tree = opts.tree
   window.tree = tree
+  var timing = opts.timing || false
   var element = opts.element
   var c = 18 // cell height
   var h = opts.height || (depth(tree) + 2) * c // graph height
   var minHeight = opts.minHeight || 950
   h = h < minHeight ? minHeight : h
   var w = opts.width || document.body.clientWidth * 0.89 // graph width
+  var scaleWidth = d3.scaleLinear().range([0, w])
   var selection = null // selection
   var transitionDuration = 500
-  var transitionEase = 'cubic-in-out' // tooltip offset
+  var transitionEase = d3.easeCubicInOut
   var sort = true
   var tiers = false
   var filterNeeded = true
   var filterTypes = []
   var allSamples
+  var focused = null
+
+  function time (name, fn) {
+    if (timing) {
+      console.time(name)
```

Comment:
are these for debugging?

Reply:
yeah all the time stuff was for debugging. it's behind the `opts.timing` option, disabled by default. we can ditch it entirely too if you prefer